torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -409,15 +409,29 @@ void TCPStore::_splitSet(
   buffer.flush();
 }
 
-void TCPStore::set(const std::string& key, const std::vector<uint8_t>& data) {
+void TCPStore::doSet(
+    const std::string& key,
+    const std::vector<uint8_t>& data,
+    bool usePrefix) {
   detail::timing_guard tguard(clientCounters_["set"]);
   const std::lock_guard<std::mutex> lock(activeOpLock_);
   detail::SendBuffer buffer(*client_, detail::QueryType::SET);
-  buffer.appendString(keyPrefix_ + key);
+  auto keyToUse = usePrefix ? keyPrefix_ + key : key;
```

Comment:
i'm confused, is `keyPrefix_` present for even basic TCPStore, and it's different from the extra prefix(es) added by PrefixStore wrappers?

What is `keyPrefix_` typically set to?

Do we need to avoid `keyPrefix_` when doing 'global' writes/reads, or is it enough to just avoid the PrefixStore prefixes?

Reply:
yes, TCPStore also use `prefix_` as well. I think it's '/'. I think we should avoid both that's why I added the method for both TCPStore and PrefixStore.

Reply:
what's your reasoning?  I think it depends if and why the prefix would be different between different TCPStore instances.  Does anyone override this prefix today?  (if so, would they expect 'globals' to be visible _across_ the instances with custom prefixes or not?)